### PR TITLE
Add connectivity framework middleware + needed patches

### PR DESCRIPTION
**Pull request by yeaissa**
Adding connectivity framework middleware.
Adding needed patches to be used with Zephyr.
Update Readme.

### CODE REVIEW

**Comment by mmahadevan108** [comment]
Why is this needed?

**Comment by yeaissa** [comment]
> Why is this needed?
@mmahadevan108 , this goes with https://github.com/zephyrproject-rtos/zephyr/pull/71613
Connectivity Framework is used for BLE support, HCI driver is using its APIs

File: mcux/middleware/mcux-sdk-middleware-connectivity-framework/platform/rw61x/fwk_platform_ble.c [hunk 1]
```diff
@@ -276,7 +276,7 @@ static void (*hci_rx_callback)(uint8_t packetType, uint8_t *data, uint16_t len);
 /*                                Public memory                               */
 /* -------------------------------------------------------------------------- */
 
-static const uint8_t hci_cal_data_params[HCI_CMD_STORE_BT_CAL_DATA_PARAM_LENGTH] = {
```
Comment (dleach02) [inline mcux/middleware/mcux-sdk-middleware-connectivity-framework/platform/rw61x/fwk_platform_ble.c]: Curious but why is this a problem?

File: mcux/middleware/mcux-sdk-middleware-connectivity-framework/FSCI/pics/FSCI_packet_structure.PNG
[the remark this answers is not in the document]
Comment (yeaissa) [reply mcux/middleware/mcux-sdk-middleware-connectivity-framework/FSCI/pics/FSCI_packet_structure.PNG:1]: Removed those files, Thanks @danieldegrasse

File: mcux/middleware/mcux-sdk-middleware-connectivity-framework/platform/rw61x/fwk_platform_ble.c (hunk 1 above)
Comment (yeaissa) [reply mcux/middleware/mcux-sdk-middleware-connectivity-framework/platform/rw61x/fwk_platform_ble.c]: This is done to be able to use those buffers inside the hci_nxp driver on Zephyr.

Comment (dleach02) [reply mcux/middleware/mcux-sdk-middleware-connectivity-framework/platform/rw61x/fwk_platform_ble.c]: Okay, I see the Zephyr PR referencing these directly so I understand. 

We should decide generally a common method for accessing these or the underlying function being done with these so we don't have to patch this each time we update the Zephyr NXP HAL.

Comment (axelnxp) [reply mcux/middleware/mcux-sdk-middleware-connectivity-framework/platform/rw61x/fwk_platform_ble.c]: @dleach02 yes this is something we want to address, we'll check how we can backport this patch to the framework.

File: mcux/middleware/mcux-sdk-middleware-connectivity-framework/SW-Content-Register.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Release Name: MCUXpresso Software Development Kit (SDK) Framework Middleware
+Release Version: 6.1.0
+Package License: LA_OPT_NXP_Software_License.htm - Additional Distribution License, Section 2.3 applies
```
Comment (danieldegrasse) [inline mcux/middleware/mcux-sdk-middleware-connectivity-framework/SW-Content-Register.txt:3]: Is this framework licensed under LA_OPT, or BSD-3? All the files seem licensed under BSD-3, but the LA_OPT license is pulled in and referenced here. The LA_OPT license isn't compatible with Zephyr, so we need to clear this up

Comment (axelnxp) [reply mcux/middleware/mcux-sdk-middleware-connectivity-framework/SW-Content-Register.txt:3]: I think we can remove it as all files are under BSD indeed, @yeaissa will address.
FYI, mcux-sdk folder seems to contain the LA_OPT license which is probably problematic according to your comment:
https://github.com/zephyrproject-rtos/hal_nxp/tree/master/mcux/mcux-sdk

Comment (danieldegrasse) [reply mcux/middleware/mcux-sdk-middleware-connectivity-framework/SW-Content-Register.txt:3]: Thanks for the heads up- it is probably problematic, yes. I'll discuss this internally and see what we can do there.

Comment (danieldegrasse) [reply mcux/middleware/mcux-sdk-middleware-connectivity-framework/SW-Content-Register.txt:3]: After checking with @yvanderv, I've confirmed we should drop this file from this PR as well as the MCUX SDK side

**Review by danieldegrasse — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Pull request merged**